### ISSUE TITLE
Unhandled exception from internal delete operation

### ISSUE THREAD

**Issue by fsigalov** [Azure/azure-webjobs-sdk#645]
I have a  webjob with a single blobTrigger function and poison handler with a queue trigger
  `public static void BlobToBlob([BlobTrigger("$logs/blob/2016/03/{name}")] Stream stream)` 
  `public static void PoisonErrorHandler([QueueTrigger("webjobs-blogtrigger-poison")] BlobTriggerPosionMessage message, TextWriter log)`
Note this repros with or without the poison handler.
The webJob is set to onDemand and is invoked by Azure scheduler every 3 minutes. About 50% of the time the job runs successfully. But the other half it fails with a 404 on a background thread.
Some of the same blobs keep getting triggered, even though they are processed successfully. 

[03/04/2016 00:19:22 > 07c9a3: SYS INFO] Status changed to Initializing
[03/04/2016 00:19:22 > 07c9a3: SYS INFO] Run script 'AzureLogReader.exe' with script host - 'WindowsScriptHost'
[03/04/2016 00:19:22 > 07c9a3: SYS INFO] Status changed to Running
[03/04/2016 00:19:25 > 07c9a3: INFO] Found the following functions:
[03/04/2016 00:19:25 > 07c9a3: INFO] AzureLogReaderWebJob.Functions.BlobToBlob
[03/04/2016 00:19:25 > 07c9a3: INFO] AzureLogReaderWebJob.Functions.PoisonErrorHandler
[03/04/2016 00:19:25 > 07c9a3: INFO] Job host started
[03/04/2016 00:21:07 > 07c9a3: INFO] Executing: 'Functions.BlobToBlob' - Reason: 'New blob detected: $logs/blob/2016/03/03/1900/000003.log'
[03/04/2016 00:21:07 > 07c9a3: INFO] Executing: 'Functions.BlobToBlob' - Reason: 'New blob detected: $logs/blob/2016/03/03/1900/000002.log'
[03/04/2016 00:21:15 > 07c9a3: INFO] Executed: 'Functions.BlobToBlob' (Succeeded)
[03/04/2016 00:21:17 > 07c9a3: INFO] Executed: 'Functions.BlobToBlob' (Succeeded)
[03/04/2016 00:21:18 > 07c9a3: ERR ] 
[03/04/2016 00:21:18 > 07c9a3: ERR ] Unhandled Exception: 
[03/04/2016 00:21:18 > 07c9a3: ERR ] Unhandled Exception: Microsoft.WindowsAzure.Storage.StorageException: The remote server returned an error: (404) Not Found. ---> System.Net.WebException: The remote server returned an error: (404) Not Found.
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at Microsoft.WindowsAzure.Storage.Shared.Protocol.HttpResponseParsers.ProcessExpectedStatusCodeNoException[T](HttpStatusCode expectedStatusCode, HttpStatusCode actualStatusCode, T retVal, StorageCommandBase`1 cmd, Exception ex)
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at Microsoft.WindowsAzure.Storage.Blob.CloudBlob.<DeleteBlobImpl>b__29(RESTCommand`1 cmd, HttpWebResponse resp, Exception ex, OperationContext ctx)
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at Microsoft.WindowsAzure.Storage.Core.Executor.Executor.EndGetResponse[T](IAsyncResult getResponseResult)
[03/04/2016 00:21:18 > 07c9a3: ERR ]    --- End of inner exception stack trace ---
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at Microsoft.WindowsAzure.Storage.Core.Executor.Executor.EndExecuteAsync[T](IAsyncResult result)
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at Microsoft.WindowsAzure.Storage.Core.Util.AsyncExtensions.<>c__DisplayClass4.<CreateCallbackVoid>b__3(IAsyncResult ar)
[03/04/2016 00:21:18 > 07c9a3: ERR ] --- End of stack trace from previous location where exception was thrown ---
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at Microsoft.Azure.WebJobs.Host.Protocols.PersistentQueueWriter`1.<DeleteAsync>d__6.MoveNext()
[03/04/2016 00:21:18 > 07c9a3: ERR ] --- End of stack trace from previous location where exception was thrown ---
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at Microsoft.Azure.WebJobs.Host.Loggers.CompositeFunctionInstanceLogger.<DeleteLogFunctionStartedAsync>d__e.MoveNext()
[03/04/2016 00:21:18 > 07c9a3: ERR ] --- End of stack trace from previous location where exception was thrown ---
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.<TryExecuteAsync>d__1.MoveNext()
[03/04/2016 00:21:18 > 07c9a3: ERR ] --- End of stack trace from previous location where exception was thrown ---
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at Microsoft.Azure.WebJobs.Host.Executors.TriggeredFunctionExecutor`1.<TryExecuteAsync>d__0.MoveNext()
[03/04/2016 00:21:18 > 07c9a3: ERR ] --- End of stack trace from previous location where exception was thrown ---
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at Microsoft.Azure.WebJobs.Host.Blobs.Listeners.BlobQueueTriggerExecutor.<ExecuteAsync>d__4.MoveNext()
[03/04/2016 00:21:18 > 07c9a3: ERR ] --- End of stack trace from previous location where exception was thrown ---
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at Microsoft.Azure.WebJobs.Host.Queues.Listeners.QueueListener.<ProcessMessageAsync>d__11.MoveNext()
[03/04/2016 00:21:18 > 07c9a3: ERR ] --- End of stack trace from previous location where exception was thrown ---
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at Microsoft.Azure.WebJobs.Host.Timers.BackgroundExceptionDispatcher.<>c__DisplayClass1.<Throw>b__0()
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
[03/04/2016 00:21:18 > 07c9a3: ERR ]    at System.Threading.ThreadHelper.ThreadStart()
[03/04/2016 00:21:18 > 07c9a3: SYS INFO] Status changed to Failed
[03/04/2016 00:21:18 > 07c9a3: SYS ERR ] Job failed due to exit code -532462766

**Comment by mathewc** [comment]
Thanks for reporting. Looks like some bug on our end that we'll need to try to get a repro for to investigate.

**Comment by JonCSL** [comment]
Just wanted to add that we get the same problem.  The webjob processes the first job on the queue fine, but then gets the above error when it completes (presumable when it tries to delete the job).  Our webjob is running on premise (i.e. not hosted in Azure).  It works fine on dev machines but not on the server which is running Windows 2012 R2.

We get this in both version 1.1.0 and 1.1.1 ( I rolled back to 1.1.0 so I could upgrade WindowsAzure.Storage to v7 to see if that fixed it but it made no difference)

Here is the exception stack:

```
Unhandled Exception: Microsoft.WindowsAzure.Storage.StorageException: The remote server returned an error: (404) Not Found. ---> System.Net.WebException: The remote server returned an error: (404) Not Found.
   at Microsoft.WindowsAzure.Storage.Shared.Protocol.HttpResponseParsers.ProcessExpectedStatusCodeNoException[T](HttpStatusCode expectedStatusCode, HttpStatusCode actualStatusCode, T retVal, StorageCommandBase`1 cmd, Exception ex)
   at Microsoft.WindowsAzure.Storage.Blob.CloudBlob.<DeleteBlobImpl>b__29(RESTCommand`1 cmd, HttpWebResponse resp, Exception ex, OperationContext ctx)
   at Microsoft.WindowsAzure.Storage.Core.Executor.Executor.EndGetResponse[T](IAsyncResult getResponseResult)
   --- End of inner exception stack trace ---
   at Microsoft.WindowsAzure.Storage.Core.Executor.Executor.EndExecuteAsync[T](IAsyncResult result)
   at Microsoft.WindowsAzure.Storage.Blob.CloudBlob.EndDelete(IAsyncResult asyncResult)
   at Microsoft.WindowsAzure.Storage.Core.Util.AsyncExtensions.<>c__DisplayClass4.<CreateCallbackVoid>b__3(IAsyncResult ar)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.WebJobs.Host.Protocols.PersistentQueueWriter`1.<DeleteAsync>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.WebJobs.Host.Loggers.CompositeFunctionInstanceLogger.<DeleteLogFunctionStartedAsync>d__e.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.<TryExecuteAsync>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.WebJobs.Host.Executors.TriggeredFunctionExecutor`1.<TryExecuteAsync>d__0.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.WebJobs.Host.Queues.Listeners.QueueTriggerExecutor.<ExecuteAsync>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.WebJobs.Host.Queues.Listeners.QueueListener.<ProcessMessageAsync>d__11.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Azure.WebJobs.Host.Timers.BackgroundExceptionDispatcher.<>c__DisplayClass1.<Throw>b__0()
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
```

**Comment by JonCSL** [comment]
Just an update - if I change the settings of the webjob that crashes to read from a queue in a different Azure account it works fine, so I guess there is something wrong with the settings on my main Azure account.  Having said that, the same webjob running on other machines can read and delete from the main Azure account just fine.

Let me know if there is any other information you need to track the problem down

**Comment by christopheranderson** [comment]
We're planning on releasing a new version of our SDK with the latest Storage SDK. Could you try that when we've got the latest pre-release out, and see if it repros then?

**Comment by JonCSL** [comment]
Yep no problem.  I assume you mean when the new pre-release version of the Microsoft.Azure.WebJob?  Do you know when it will be?

**Comment by mathewc** [comment]
You can already get the pre-release versions on MyGet, instructions here: https://github.com/Azure/azure-webjobs-sdk/wiki/%22Nightly%22-Builds. We'll get up on Nuget soon.

**Comment by mathewc** [comment]
FYI: We've released the [2.0.0-beta1](https://www.nuget.org/packages/Microsoft.Azure.WebJobs/2.0.0-beta1) prerelease on Nuget. Please give that a try and see if you're still having issues.

**Comment by lindydonna** [comment]
Closing since we didn't get a response. Please reopen if you can repro with the latest beta1 bits.
